domain/goal/src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/GoalServiceTest.kt

```diff
@@ -17,13 +17,18 @@ import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.aValidGo
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.dto.aValidCreateGoalDto
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.dto.aValidUpdateGoalDto
 
-@ExtendWith(MockitoExtension::class)
+@TestInstance(Lifecycle.PER_CLASS)
 class GoalServiceTest {
-  @InjectMocks
   private lateinit var service: GoalService
 
-  @Mock
-  private lateinit var persistenceAdapter: GoalPersistenceAdapter
+  private var persistenceAdapter: GoalPersistenceAdapter = mock()
+
+  private var goalServiceDecorator: GoalServiceDecorator = mock()
+
+  @BeforeAll
+  fun setupService() {
+    service = GoalService(persistenceAdapter, goalServiceDecorator)
+  }
```

Comment:
Because the `GoalService` has an optional constructor arg (for the decorator), as far as the java byte code is concerned the class has 2 constructors (one for just the persistence adapter, and one for the persistence adapter and the decorator)
Because there are 2 constructors, InjectMocks doesn't work and mockito complains bitterly that it doesnt know which constructor to use! 
Therefore we have to construct the instance manually

Reply:
Minor, but I think this is the way we'd normally address this:

```
@ExtendWith(MockitoExtension::class)
class GoalServiceTest {
  private lateinit var service: GoalService

  @Mock
  private lateinit var persistenceAdapter: GoalPersistenceAdapter

  @Mock
  private lateinit var goalServiceDecorator: GoalServiceDecorator

  @BeforeEach
  fun setupService() {
    service = GoalService(persistenceAdapter, goalServiceDecorator)
  }
```

It's not necessarily better and in the absence of any internal state within the service class, you could argue that `Lifecycle.PER_CLASS` is more efficient, but it's a smaller change :)